pkgs/development/compilers/p4c/default.nix

```diff
@@ -0,0 +1,89 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boehmgc
+, bison
+, flex
+, protobuf
+, gmp
+, boost
+, python3
+, doxygen
+, graphviz
+, libbpf
+, libllvm
+, enableDocumentation ? true
+, enableBPF ? true
+, enableDPDK ? true
+, enableBMV2 ? true
+, enableGraphBackend ? true
+, enableP4Tests ? true
+, enableGTests ? true
+, enableMultithreading ? false
+}:
+let
+  toCMakeBoolean = v: if v then "ON" else "OFF";
+in
+stdenv.mkDerivation rec {
+  pname = "p4c";
+  version = "1.2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "p4lang";
+    repo = "p4c";
+    rev = "v${version}";
+    sha256 = "sha256-XIZ7Cm5zfr5XA+s0aSG1WwWHCPjAYv/YoBWTRaXi9rQ=";
+    fetchSubmodules = true;
+  };
+
+  postFetch = ''
+    rm -rf backends/ebpf/runtime/contrib/libbpf
+    rm -rf control-plane/p4runtime
+  '';
+
+  cmakeFlags = [
+    "-DENABLE_BMV2=${toCMakeBoolean enableBMV2}"
+    "-DENABLE_EBPF=${toCMakeBoolean enableBPF}"
+    "-DENABLE_UBPF=${toCMakeBoolean enableBPF}"
+    "-DENABLE_DPDK=${toCMakeBoolean enableDPDK}"
+    "-DENABLE_P4C_GRAPHS=${toCMakeBoolean enableGraphBackend}"
+    "-DENABLE_P4TEST=${toCMakeBoolean enableP4Tests}"
+    "-DENABLE_DOCS=${toCMakeBoolean enableDocumentation}"
+    "-DENABLE_GC=ON"
+    "-DENABLE_GTESTS=${toCMakeBoolean enableGTests}"
+    "-DENABLE_PROTOBUF_STATIC=ON"
+    "-DENABLE_MULTITHREAD=${toCMakeBoolean enableMultithreading}"
+    "-DENABLE_GMP=ON"
+  ];
+
+  checkTarget = "make check";
+
+  nativeBuildInputs = [
+    bison
+    flex
+
+    cmake
+
+    python3
+
+    protobuf
+  ] ++ lib.optional enableBPF libbpf;
+
+  buildInputs = [
+    boehmgc
+    boost
+    gmp
+  ]
+  ++ lib.optional enableDocumentation [ doxygen graphviz ]
```

Comment:
Those most likely also go to nativeBuildInputs. Try setting ``strictDpes = true;`` and you should notice that the build will fail.

Reply:
Let me try.

Reply:
Using `strictDeps = true;`, well, protobuf is not detected anymore, which is in `nativeBuildInputs`, am I right to infer that should rather be in `buildInputs` in that case?

Reply:
Okay, I tried to put most of the buildInputs in the nativeBuildInputs, and it did not work. Also, with `strictDeps`, I'm now failing with error 2 without any relevant log, I'm investigating.

Reply:
I think protobuf is not that easy because it can be used as a cli and library but I didn't double check that for this program.